src/tools/kdb/complete.cpp

```diff
@@ -102,6 +106,17 @@ void CompleteCommand::addMountpoints (KeySet & ks, const Key root)
 	printWarnings (cerr, mountpointPath);
 }
 
+void CompleteCommand::addNamespaces (map<Key, pair<int, int>> & hierarchy)
+{
+	// Currently assumed to be fixed, and they are on level 0
+	const string namespaces[] = { "spec/", "proc/", "dir/", "user/", "system/" };
```

Comment:
is there a way to get all available namespaces, because i haven't found any? They only thing i've found are integer constants e.g. KEY_NS_DIR, but those can't really be translated back to a name like "dir/" right?

keyGetNamespace can be used to get a key's namespace yes, but what if there is nothing in e.g. proc yet, then even by iterating all keys and checking their namespace, we wouldn't be able to get all the available namespaces. 

So if i add such method to the c++ bindings (though it might be even better to have such method on a lower level as its not really binding specific) wouldn't it end up returning a hardcoded list of namespaces?